pkg/rotator/rotator.go

```diff
@@ -692,6 +711,19 @@ func (r *ReconcileWH) Reconcile(ctx context.Context, request reconcile.Request)
 	}
 
 	if secret.GetDeletionTimestamp().IsZero() {
+		if r.refreshCertIfNeededDelegate != nil {
+			rotatedCA, err := r.refreshCertIfNeededDelegate()
+			if err != nil {
+				crLog.Error(err, "error rotating certs on secret reconcile")
+				return reconcile.Result{}, err
+			}
+
+			// if we did rotate the CA, the secret is stale so let's return
+			if rotatedCA {
+				return reconcile.Result{}, nil
```

Comment:
why don't we need to requeue the request?

Reply:
> Re-queueing immediately means we may re-run the reconcile loop before the informer cache has been updated, which means we will be operating on a stale version of the (now valid-on-server, but invalid-in-cache) secret. This will cause us to attempt to re-rotate the secret, which will fail due to stale write errors.
>
> We should probably just return without re-queueing. When we get a watch event for the secret, the reconcile loop will run with the updated view.

https://github.com/open-policy-agent/cert-controller/pull/81#discussion_r1235986806

Reply:
thanks Max 🙏🏼 

@ritazh let us/ me know if the above didn't address your question or for any follow ups

Reply:
thanks for the explanation